wiki/Client/File_formats/Osu_(file_format)/en.md

```diff
@@ -4,346 +4,623 @@
+### Parameters
 
-- **`time` (Integer):** Start time of the timing section, in milliseconds from the beginning of the beatmap's audio. The end of the timing section is the next timing point's time (or never, if this is the last timing point).
-- **`beatLength` (Decimal):** This property has two meanings:
-  - For uninherited timing points, the duration of a beat, in milliseconds.
-  - For inherited timing points, a negative inverse slider velocity multiplier, as a percentage. For example, `-50` would make all sliders in this timing section twice as fast as `SliderMultiplier`.
-- **`meter` (Integer):** Amount of beats in a measure. Inherited timing points ignore this property.
-- **`sampleSet` (Integer):** Default sample set for hit objects (0 = beatmap default, 1 = normal, 2 = soft, 3 = drum).
-- **`sampleIndex` (Integer):** Custom sample index for hit objects. `0` indicates osu!'s default hitsounds.
-- **`volume` (Integer):** Volume percentage for hit objects.
-- **`uninherited` (0 or 1):** Whether or not the timing point is uninherited.
-- **`effects` (Integer):** Bit flags that give the timing point extra effects. See [the effects section](#effects).
+`start_time,beat_length,meter,sample_set,sample_index,volume,uninherited,effects`
 
-### Effects
+| Option | Value type | Description | Value |
+| :-- | :-- | :-- | :-- |
+| `start_time` | Integer | Time in milliseconds when the timing section should start | `12000` |
+| `beat_length` | Decimal | Time in milliseconds of the beat duration | `250` |
+| `meter` | Integer | Amount of beats in a measure | `4` |
+| `sample_set` | Integer | Default sample set for hit objects | `3` |
+| `sample_index` | Integer | Custom sample index for hit objects | `0` |
+| `volume` | Integer | Volume percentage for hit objects | `70` |
+| `uninherited` | 0 or 1 | Whether or not the timing point is uninherited | `0` |
+| `effects` | Integer | Bit flags that give the timing point extra effects | `1` |
 
-Timing points have two extra effects that can be toggled using bits 0 and 3 (from least to most significant) in the `effects` integer:
+### Examples
 
-- 0: Whether or not [kiai time](/wiki/Gameplay/Kiai_time) is enabled
-- 3: Whether or not the first barline is omitted in osu!taiko and osu!mania
+`10000,333.33,4,0,0,100,1,1`
 
-The rest of the bits are unused.
+`12000,250,4,3,0,70,0,1`
 
-### Examples
+### Notice
+
+`start_time`
+
+> The end of the timing section is the next timing point's time or never, if this is the last timing point.
+
+`beat_length`
+
+> For inherited points, it is the inverse multiplier of the slider speed as a percentage. For example, `-50` will make all sliders in this time section double as fast as `SliderMultiplier`. In example 1, `333.33` means `1 / 333.33 * 1000 * 60 = 180` BPM.
+
+`meter`
+
+> Inherited timing points ignore this property.
+
+`sample_set`
+
+> Possible values:
+>
+> - `0` - Default
```

Comment:
I just tested it and it turns out setting `sample_set` to 0, regardless if point is inherited or not, always sets sample set to `Soft` and ignores `General.SampleSet`